editor/src/components/navigator/left-pane.tsx

```diff
@@ -777,22 +784,81 @@ const GithubPane = React.memo(() => {
     [dispatch],
   )
 
-  const [githubRepoToSaveTo, setGithubRepoToSaveTo] = React.useState<string | undefined>(
+  const [targetRepository, setTargetRepository] = React.useState<string | undefined>(
     storedTargetGithubRepo,
   )
+  const parsedTargetRepository = React.useMemo(() => {
+    if (targetRepository == null) {
+      return null
+    } else {
+      return parseGithubProjectString(targetRepository)
+    }
+  }, [targetRepository])
 
-  const onChangeGithubRepoToSaveTo = React.useCallback(
+  const onChangeTargetRepository = React.useCallback(
     (changeEvent: React.ChangeEvent<HTMLInputElement>) => {
-      setGithubRepoToSaveTo(changeEvent.currentTarget.value)
+      setTargetRepository(changeEvent.currentTarget.value)
     },
-    [setGithubRepoToSaveTo],
+    [setTargetRepository],
   )
 
   const triggerSaveToGithub = React.useCallback(() => {
-    if (githubRepoToSaveTo != null) {
-      dispatch([EditorActions.saveToGithub(githubRepoToSaveTo)], 'everyone')
+    if (parsedTargetRepository != null) {
+      dispatch([EditorActions.saveToGithub(parsedTargetRepository)], 'everyone')
+    }
+  }, [dispatch, parsedTargetRepository])
+
+  const [branchesForRepository, setBranchesForRepository] =
+    React.useState<GetBranchesResponse | null>(null)
+
+  React.useEffect(() => {
+    if (parsedTargetRepository != null) {
+      void getBranchesForGithubRepository(parsedTargetRepository).then((result) => {
+        setBranchesForRepository(result)
+      })
+    }
+  }, [parsedTargetRepository])
+
+  const getBranchesUI = React.useCallback(() => {
```

Comment:
Looks like `React.useMemo` would be better suited to this

Reply:
Fixed in 8fc83dd4a.